spec/unit/network/authconfig_spec.rb

```diff
@@ -107,3 +107,29 @@
     end
   end
 end
+
+describe Puppet::Network::AuthConfig do
+  class TestAuthProvider
+    def initialize(rights=nil); end
+    def check_authorization(method, path, params); end
+  end
+
+  it "instantiates authprovider_class with rights" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    rights = Puppet::Network::Rights.new
+    TestAuthProvider.expects(:new).with(rights)
+    described_class.new(rights)
+  end
+
+  it "delegates authorization check to authprovider_class" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    TestAuthProvider.any_instance.expects(:check_authorization).with(:save, '/path/to/resource', {})
+    described_class.new.check_authorization(:save, '/path/to/resource', {})
+  end
+
+  it "uses DefaultAuthProvider by default" do
+    Puppet::Network::AuthConfig.authprovider_class = nil
```

Comment:
> Happy to go either way, not sure which one is better?

Actually I like the current implementation better than the one above in my comment.  Going to keep `AuthConfig` the way it is, though I will reorder the tests to assert the default first as it reads better and avoids the odd `= nil`.